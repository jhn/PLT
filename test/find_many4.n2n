fn main() -> Void {

	actor: Node = { name: String, age: Int }
	Keanu: Node = actor["Keanu", 35]
	Leo: Node = actor["Leo", 20]

	actedIn: Rel = { role: String }
	neo: Rel = actedIn["Neo"]
	jordan: Rel = actedIn["Jordan"]
	nelson: Rel = acted_in["Nelson"]

	movie: Node = { title: String, year: Int }
	matrix: Node = movie["Matrix", 1999]
	wolf: Node = movie["Wolf", 1994]
	sweet_nov: Node = movie["Sweet November", 2000]

	Cast: Graph = <
		Keanu neo matrix,
		Leo jordan wolf,
		Keanu nelson sweet_nov
	>

	node_lit: List<Node> 
	node_lit = Cast.find_many(actor["Keanu"])
	
	print(node_lit)

}