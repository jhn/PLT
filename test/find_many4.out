[actor{ name = Keanu, age = 35 }]
